src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -85,6 +92,124 @@ class ResultsNeedFullReloadError(RuntimeError):
+    :ivar property_id:
+        The property name (holdup, total mass flow rate).
+    :ivar trend_id:
+        The id associated a specific trend.
+    :ivar category:
+        The property category. For global sensitivity analysis
+        this always be dimensionless.
+    :ivar parametric_var_id:
+        The id associated a parametric variable.
+    :ivar parametric_var_name:
+        The name associated a parametric variable.
+    :ivar network_element_name:
+        The name of network where this data is associated.
+    :ivar position:
+        The element position in a pipe.
+    :ivar position_unit:
+        The position unit (m, Km, cm).
+    :ivar unit:
+        The unit of a dimensionless property (-, %)
+    :ivar qoi_index:
+        The index of a quantity of interest in the results file.
+    :ivar qoi_data_index:
+        The data index of a quantity of interest.
+    """
+
+    property_id: str = attr.ib(validator=attr.validators.instance_of(str))
+    trend_id: str = attr.ib(validator=attr.validators.instance_of(str))
+    category: str = attr.ib(validator=attr.validators.instance_of(str))
+    parametric_var_id: str = attr.ib(validator=attr.validators.instance_of(str))
+    parametric_var_name: str = attr.ib(validator=attr.validators.instance_of(str))
+    network_element_name: Optional[str] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(str))
+    )
+    position: Optional[float] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(float))
+    )
+    position_unit: Optional[str] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(str))
+    )
+    unit: str = attr.ib(validator=attr.validators.instance_of(str))
+    qoi_index: Optional[int] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(int))
+    )
+    qoi_data_index: Optional[int] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(int))
+    )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> Self:
+        return cls(
+            property_id=data["property_id"],
+            trend_id=data["trend_id"],
+            category=data["category"],
+            parametric_var_id=data["parametric_var_id"],
+            parametric_var_name=data["parametric_var_name"],
+            network_element_name=data["network_element_name"],
+            position=data["position"],
+            position_unit=data["position_unit"],
+            unit=data["unit"],
+            qoi_index=data["qoi_index"],
+            qoi_data_index=data["qoi_data_index"],
+        )
+
+
+@attr.s(slots=True, hash=False)
+class GlobalSensitivityAnalysisMetadata:
+    """
+    A class that hold the global sensitivity analysis metadata.
+    """
+
+    global_sensitivity_analysis_meta_data: Dict[
+        str, GlobalSensitivityAnalysisMetaInformation
+    ] = attr.ib(validator=attr.validators.instance_of(Dict))
```

Comment:
This is a bit confusing:

1. `GlobalSensitivityAnalysisMetaInformation` and `GlobalSensitivityAnalysisMetadata` are nearly synonymous, so it is hard to tell them apart just from the name, without looking at the class definition/docs.
2. `GlobalSensitivityAnalysisMetadata.global_sensitivity_analysis_meta_data` is a dict of `GlobalSensitivityAnalysisMetaInformation`... this should send warning signals to you, having a class variable which is the name of the class itself, but references actually another data type.

I think `GlobalSensitivityAnalysisMetadata` is fine, but we should rename `GlobalSensitivityAnalysisMetaInformation` to something else. Doesn't it hold information about one of the parametric variables in a Global SA? How about `ParametricVariableMetadata`? If so, rename the `global_sensitivity_analysis_meta_data` as well to reflect that, `parametric_variables_metadata`.

Reply:
I renamed it to ```QuantityOfInterestMetaData``` that what it really represent.

Reply:
Sounds good!

Reply:
Humm, i agree.

Reply:
No I was not aware, I agree an intermediate "item" container would be good.

Reply:
Actually I'm not sure I follow, but I will defer to your decision here.